### PR TITLE
BEMXJST: applyCtx fixes

### LINKED ISSUE #202: applyCtx brokes this.position
Example: http://goo.gl/fcnGpn

### PR DESCRIPTION
**Pull request by miripiruni**
Fixes #202

### CODE REVIEW

File: test/bemcontext-position-test.js [hunk 1]
```diff
@@ -178,4 +178,19 @@ describe('BEMContext this.position', function() {
       '<div class="b b_last_yes p_3"></div>' +
     '</div>');
   });
+
+  it('should properly set position with applyCtx()', function() {
+    test(function() {
+      block('a')(
+        def()(function() {
+        applyCtx({ block: 'session' });
```
Comment (zxqfox) [inline test/bemcontext-position-test.js]: отступы сломались :(

File: lib/bemxjst/index.js [hunk 2]
```diff
@@ -50,6 +50,10 @@ BEMXJST.prototype.compile = function(code) {
   var self = this;
 
   function applyCtx() {
+    return self.run(self.context.ctx);
```
Comment (miripiruni) [inline lib/bemxjst/index.js]: @zxqfox see in context of #203 

`run()` — create new BEMContext
`_run()` — doesnt

**Comment by miripiruni** [comment]
@zxqfox @tadatuta I just updated PR. Fixed #202 and #203

File: test/runtime-applyctx-test.js [hunk 3]
```diff
@@ -29,4 +29,21 @@ describe('Runtime applyCtx()', function() {
     { block: 'b1' },
     '<div class="b1">&lt;div class="b2"&gt;ohai&lt;/div&gt;</div>');
   });
+
+  it('should not affect context if star template', function() {
```
Comment (zxqfox) [inline test/runtime-applyctx-test.js]: should not affect? what does this mean? ;-D

File: test/runtime-applyctx-test.js [hunk 4]
```diff
@@ -29,4 +29,21 @@ describe('Runtime applyCtx()', function() {
...
+          applyCtx({ block: 'b' });
+          return applyNext();
+        })
+      );
+    },
+    [
+        { block: 'a' },
+        { block: 'a' }
+    ],
+    '<div class="b"></div><div class="a"></div>');
```
Comment (zxqfox) [inline test/runtime-applyctx-test.js]: I don't see how this was changed.

**Comment by zxqfox** [comment]
:shipit: 

cc @tadatuta

**Pull request merged**